chain/substreams/src/trigger.rs

```diff
@@ -85,3 +100,133 @@ impl blockchain::TriggersAdapter<Chain> for TriggersAdapter {
+                                    "Invalid field type, the protobuf probably needs updating"
+                                )))
+                            }
+                            (Type::Bigdecimal, false) => {
+                                match BigDecimal::parse_bytes(field.new_value.as_ref()) {
+                                    Some(bd) => Value::BigDecimal(bd),
+                                    None => {
+                                        return Err(MappingError::Unknown(anyhow!(
+                                            "Unable to parse BigDecimal for entity {}",
+                                            entity_change.entity
+                                        )))
+                                    }
+                                }
+                            }
+                            (Type::Bigint, false) => Value::BigInt(BigInt::from_signed_bytes_be(
+                                field.new_value.as_ref(),
+                            )),
+                            (Type::Int, false) => {
+                                let mut bytes: [u8; 4] = [0; 4];
+                                bytes.copy_from_slice(field.new_value.as_ref());
+                                Value::Int(i32::from_be_bytes(bytes))
+                            }
+                            (Type::Bytes, false) => {
+                                Value::Bytes(Bytes::from(field.new_value.as_ref()))
+                            }
+                            (Type::String, false) => Value::String(
+                                String::from_utf8(field.new_value.clone())
+                                    .map_err(|e| MappingError::Unknown(anyhow::Error::from(e)))?,
+                            ),
+                            _ => {
+                                return Err(MappingError::Unknown(anyhow!(
+                                    "received substream new_value null, type {} field {}",
+                                    entity_change.entity,
+                                    field.name
+                                )))
+                            }
+                        };
+                        data.entry(field.name.clone()).or_insert(value);
+                    }
+
+                    write_poi_event(
+                        proof_of_indexing,
+                        &ProofOfIndexingEvent::SetEntity {
+                            entity_type: &entity_type,
+                            id: &entity_id,
+                            data: &data,
+                        },
+                        causality_region,
+                        logger,
+                    );
+
+                    state.entity_cache.set(key, Entity::from(data))?;
+                }
+                Operation::Delete => {
+                    let key = EntityKey {
+                        subgraph_id: self.locator.hash.clone(),
+                        entity_type: EntityType::new("some-type".into()),
```

Comment:
Are missing a real value here? If unused, I would add a comment explaining why we do this.

Reply:
Yep